### PR TITLE
Move annotating list to new utils.containers

### PR DESCRIPTION
**Pull request by bartvm**

### CODE REVIEW

**Comment by bartvm** [comment]
The `DefaultOrderedDict` from #476 could go here as well eventually.

File: blocks/utils/containers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    def __eq__(self, other):
+        return self._items == other
+
+    def __ne__(self, other):
+        return self._items != other
+
+    def __getitem__(self, key):
+        return self._items[key]
+
+    def _setitem(self, key, value):
```
Comment (dwf) [inline blocks/utils/containers.py]: `@abstractmethod`?

File: blocks/utils/containers.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        """The operation to perform when an item is inserted/appended."""
+        pass
+
+    def _delitem(self, key):
```
Comment (dwf) [inline blocks/utils/containers.py]: ditto, `@abstractmethod`?

Comment (dwf) [inline blocks/utils/containers.py]: Actually I can see this one perhaps not being annotated as `@abstractmethod`, but `_setitem` seems fundamental.

Comment (bartvm) [inline blocks/utils/containers.py]: Yeah, don't think this one should, sometimes you only want to perform an action on insert. I figured there theoretically could be a case where you only want to do something on deletion, which is why I didn't annotate the `_setitem` method, but I can't actually think of an example, so I guess I might as well :)

**Comment by bartvm** [comment]
@dwf Updated and rebased

**Comment by dwf** [comment]
It seems like we're losing coverage by a thousand papercuts, but as long as it's still above 91% I'm not going to worry too much about it.

**Pull request merged**